server/dbInit.ts

```diff
@@ -1,6 +1,165 @@
 import axios from 'axios';
+import shortid from 'shortid';
 import { Classes, Subjects } from './dbDefs';
 
+export const defaultEndpoint = "https://classes.cornell.edu/api/2.0/";
+
+// Represents a subject which is scraped
+// Note: there's a load of additional information when we scrape it.
+// It's not relevant, so we just ignore it for now.
+export interface ScrapingSubject {
+  descrformal: string; // Subject description, e.g. "Asian American Studies"
+  value: string; // Subject code, e.g. "AAS"
+}
+
+// Represents a class which is scraped
+// Note: there's a load of additional information when we scrape it.
+// It's not relevant, so we just ignore it for now.
+export interface ScrapingClass {
+  subject: string; // Short: e.g. "CS"
+  catalogNbr: string; // e.g. 1110
+  titleLong: string; // long variant of a title e.g. "Introduction to Computing Using Python"
+}
+
+/*
+ * Fetch the class roster for a semester.
+ * Returns the class roster on success, or null if there was an error.
+ */
+export async function fetchSubjects(endpoint: string, semester: string): Promise<ScrapingSubject[]> {
+  const result = await axios.get(`${endpoint}config/subjects.json?roster=${semester}`, { timeout: 10000 });
+  if (result.status !== 200 || result.data.status !== "success") {
+    console.log(`Error fetching ${semester} subjects! HTTP: ${result.statusText} SERV: ${result.data.status}`);
+    return null;
+  }
+
+  return result.data.data.subjects;
+}
+
+/*
+ * Fetch all the classes for that semester/subject combination
+ * Returns a list of classes on success, or null if there was an error.
+ */
+export async function fetchClassesForSubject(endpoint: string, semester: string, subject: ScrapingSubject): Promise<ScrapingClass[]> {
+  const result = await axios.get(`${endpoint}search/classes.json?roster=${semester}&subject=${subject.value}`, { timeout: 10000 });
+  if (result.status !== 200 || result.data.status !== "success") {
+    console.log(`Error fetching subject ${semester}-${subject.value} classes! HTTP: ${result.statusText} SERV: ${result.data.status}`);
+    return null;
+  }
+
+  return result.data.data.classes;
+}
+
+/*
+ * Fetch the relevant classes, and add them to the collections
+ * Returns true on success, and false on failure.
+ */
+export async function fetchAddCourses(endpoint: string, semester: string): Promise<boolean> {
+  const subjects = await fetchSubjects(endpoint, semester);
+
+  const v1 = await Promise.all(subjects.map(async (subject) => {
+    const subjectIfExists = await Subjects.findOne({ subShort: subject.value.toUpperCase() }).exec();
+
+    if (!subjectIfExists) {
+      console.log(`Adding new subject: ${subject.value}`);
+      const res = await new Subjects({
+        _id: shortid.generate(),
+        subShort: subject.value.toUpperCase(),
+        subFull: subject.descrformal,
+      }).save().catch((err) => {
+        console.log(err);
```

Comment:
Same idea as above, helpful to preempt the error you print, like this:
https://github.com/cornell-dti/course-reviews-react-2.0/blob/f30c4ca2f22682e574e47f703c3fd1100d558f6b/server/methods.ts#L198